src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,203 @@
+            if (resultSet.next()) {
+                car = getCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car by id " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversFromCar(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String getAllCarQuery = "SELECT c.id AS car_id, model, "
+                + "manufacturer_id, m.name, m.country "
+                + "FROM cars c JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllStatement
+                        = connection.prepareStatement(getAllCarQuery)) {
+            ResultSet resultSet = getAllStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get a list of cars "
+                    + "from cars table.", e);
+        }
+        cars.forEach(car -> car.setDrivers(getDriversFromCar(car.getId())));
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery = "UPDATE cars SET manufacturer_id = ?, model = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateStatement = connection.prepareStatement(updateQuery)) {
+            updateStatement.setLong(1, car.getManufacturer().getId());
+            updateStatement.setString(2, car.getModel());
+            updateStatement.setLong(3, car.getId());
+            updateStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Couldn't update " + car + " in cars DB.", e);
+        }
+        deleteDrivers(car);
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteQuery = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteStatement = connection.prepareStatement(deleteQuery)) {
+            deleteStatement.setLong(1, id);
+            return deleteStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllByDriverQuery = "SELECT c.id AS car_id, model, m.id AS manufacturer_id, "
+                + "m.name, m.country FROM cars c JOIN cars_drivers cd ON c.id = cd.car_id "
```

Comment:
Let's move  `FROM` and `JOIN` to the new line for the better code readability:
```suggestion
                + "m.name, m.country" 
                + "FROM cars c" 
                + "JOIN cars_drivers cd ON c.id = cd.car_id "
```